lib/screens/splash/splash_screen.dart

```diff
@@ -15,9 +16,10 @@ class SplashScreen extends StatelessWidget {
         listenWhen: (prevState, state) => prevState.status != state.status,
         listener: (context, state) {
           if (state.status == AuthStatus.unauthenticated) {
-            Navigator.of(context).pushNamed(LoginScreen.routeName);
+            Navigator.of(context)
+                .pushReplacementNamed(NewLoginScreen.routeName);
           } else if (state.status == AuthStatus.authenticated) {
-            Navigator.of(context).pushNamed(NavScreen.routeName);
+            Navigator.of(context).pushReplacementNamed(NavScreen.routeName);
```

Comment:
I feel that we should not replace the splash screen, since this is the screen that is listening to the changes in the authState and changing between nav and login screen.

I feel this code is the reason your code was not able to change between nav and login screens automatically.